server/src/creators/thread-creator.js

```diff
@@ -86,13 +90,18 @@ async function createThread(
     creatorChangeset,
     initialMembersChangeset,
     recalculatePermissionsChangeset,
+    knowOfRelationshipsChangeset,
   ] = await Promise.all([
     changeRole(id, [viewer.userID], newRoles.creator.id),
     initialMemberIDs && initialMemberIDs.length > 0
       ? changeRole(id, initialMemberIDs, null)
       : undefined,
     recalculateAllPermissions(id, threadType),
+    initialMemberIDs && initialMemberIDs.length > 0
+      ? knowOfRelationshipsForThread(initialMemberIDs.concat(viewer.userID))
+      : undefined,
```

Comment:
There is some complexity here that I realized I never explained. Ultimately I think we will need to handle `knowOfRelationshipsForThread` inside `thread-permission-updaters`.

Here's the issue:
1. Let's imagine Bob is trying to create a new child thread. The parent thread has users Bob, Alice, and Jane. For the child thread, Bob adds Alice, but also a new user Joe.
2. Because Joe is now in a child thread that has access to the parent thread, the server will create a `membership` row for the parent thread, and pass that parent thread down to Joe. Joe's `role` for this `membership` row in the DB will be `0`, which is represented as `null` in the `MemberInfo` type in JS. This value indicates Joe is not truly a member of the parent thread, and it's similar in concept to `know_of` - the server needs to make sure Joe has an up-to-date version of the parent thread, but Joe is not technically a member.
3. Since Joe has access to the parent thread, we now need to create a `know_of` row between Joe and Jane. But we won't know that we need this row unless we look at the parent thread's members. Furthermore, since threads can be infinitely nested, this problem requires recursively looking up all the parents until we get to the root.

`thread-permission-updaters` has to handle a very similar set of problems, and ends up recursing to the root thread anyways when it's creating a new thread. I also think it would be a good place to put this `knowOfRelationshipsForThread` logic because all membership changes have to go through there anyways. I'm not 100% sure, but I think you could probably avoid having to make any changes to `thread-creator` or `thread-updaters` if you handle everything in `thread-permission-updaters`.